core/state/access_witness.go

```diff
@@ -20,391 +20,235 @@ import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/params"
 	"github.com/ethereum/go-ethereum/trie/utils"
+	"github.com/holiman/uint256"
 )
 
-type VerkleStem [31]byte
-
-// Mode specifies how a tree location has been accessed
+// mode specifies how a tree location has been accessed
 // for the byte value:
 // * the first bit is set if the branch has been edited
 // * the second bit is set if the branch has been read
-type Mode byte
+type mode byte
 
 const (
-	AccessWitnessReadFlag  = Mode(1)
-	AccessWitnessWriteFlag = Mode(2)
+	AccessWitnessReadFlag  = mode(1)
+	AccessWitnessWriteFlag = mode(2)
 )
 
+var zeroTreeIndex uint256.Int
+
 // AccessWitness lists the locations of the state that are being accessed
 // during the production of a block.
 type AccessWitness struct {
-	// Branches flags if a given branch has been loaded
-	Branches map[VerkleStem]Mode
-
-	// Chunks contains the initial value of each address
-	Chunks map[common.Hash]Mode
-
-	// InitialValue contains either `nil` if the location
-	// didn't exist before it was accessed, or the value
-	// that a location had before the execution of this
-	// block.
-	InitialValue map[string][]byte
-
-	// Caches which code chunks have been accessed, in order
-	// to reduce the number of times that GetTreeKeyCodeChunk
-	// is called.
-	CodeLocations map[string]map[uint64]struct{}
+	branches map[branchAccessKey]mode
+	chunks   map[chunkAccessKey]mode
```

Comment:
These are the new tracking maps. Each of them track preimages and not Pedersen Hashes.
See the key-type (end of this file) to understand what we use as keys. 

They're carefully created to being able to be used as maps in Go (since you can't use `[]byte` etc), and I wanted a single map nesting (so avoid `map[...]map[...]` which is tricky and messy)

Reply:
On the contrary, it should cause less than the previous nested maps.